### PR TITLE
Add recipe for mxnet

### PR DESCRIPTION
**Pull request by nehaljwani**
- This is for *nix only.
- The build on macOS hasn't been compiled with openmp
- Since we don't have an up to date version of {python-,}graphviz at the moment, installing {python-,}graphviz in the same environment will not be possible, as they will conflict with the dependencies of the latest opencv with which this package will be built. To work around that, we'll need to build it with opencv 3.1.0 and opencv 3.3.1

I spent some time to include the nomkl variant, but it is quite hard to get it right with split packages at the moment, so I'll submit a separate recipe for that, with the feedstock name: mxnet-nomkl

### CODE REVIEW

File: mxnet-mkl-feedstock/recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        - numpy >=1.8.2,<=1.13.3
+        - requests >=2.18.4,<3
+        # This is only required for visualization and hence optional.
+        # We need a new build of graphviz otherwise it downgrades opencv
+        #- python-graphviz
+    test:
+      imports:
+        - mxnet
+
+  - name: mxnet-cpp
```
Comment (mingwandroid) [inline mxnet-mkl-feedstock/recipe/meta.yaml]: What is this package? IMHO there should be no packages ending with `-cpp`. For the following reasons:

1. `conda-forge` thought it stood for C++, but it does not. It stands for C preprocessor.
2. We use prefixes not suffixes to indicate package namespaces (though I'm not *really* sure that's your intention?)
3. It seems this is just C headers which indicates that you might mean `cpp` to stand for `C preprocessor`? Regardless, due to CF, these waters are too muddied.
4. Mostly, why bother? Headers are tiny and should be bundled with `libmxnet` where they are most useful and can most reasonably be expected to be found.

Comment (nehaljwani) [reply mxnet-mkl-feedstock/recipe/meta.yaml]: The upstream folks maintain it as a separate package.
https://github.com/apache/incubator-mxnet/tree/master/cpp-package

What should we call it? Should we still merge it with libmxnet?

Comment (nehaljwani) [reply mxnet-mkl-feedstock/recipe/meta.yaml]: Hmm.. the headers are inside mxnet-cpp subfolder anyway, so I'll just combine them in libmxnet.

**Comment by nehaljwani** [comment]
Recipe updated. I've merged the mxnet-cpp headers into libmxnet

**Comment by nehaljwani** [comment]
For Windows, I'll probably wait till 1.0.1 is released. Our opencv package needs to be fixed anyway.

**Comment by nehaljwani** [comment]
Finally, this is ready for both: mkl and nomkl.

**Comment by nehaljwani** [comment]
Trying to add jemalloc support now.

**Comment by nehaljwani** [comment]
@mingwandroid This is ready now.

**Pull request merged**